src/index.js

```diff
@@ -17,62 +17,72 @@
-const Replacer = require('./replacer');
-const Reveal = require('./reveal');
-const Retest = require('./retest');
-const Revisit = require('./revisit');
-const RuleConfig = require('./ruleConfig')
-const Script = require('./script');
-const Search = require('./search');
-const Selenium = require('./selenium');
-const SessionManagement = require('./sessionManagement');
-const Soap = require('./soap');
-const Spider = require('./spider');
-const Stats = require('./stats');
-const Users = require('./users');
-const Wappalyzer = require('./wappalyzer');
-const Websocket = require('./websocket');
+
+// const request = require("request");
+// const requestPromise = require("request-promise-native");
+const axios = require("axios");
+const AccessControl = require("./accessControl");
+const Acsrf = require("./acsrf");
+const AjaxSpider = require("./ajaxSpider");
+const Alert = require("./alert");
+const AlertFilter = require("./alertFilter");
+const Ascan = require("./ascan");
+const Authentication = require("./authentication");
+const Authorization = require("./authorization");
+const Automation = require("./automation");
+const Autoupdate = require("./autoupdate");
+const Brk = require("./brk");
+const Context = require("./context");
+const Core = require("./core");
+const Exim = require("./exim");
+const ForcedUser = require("./forcedUser");
+const Graphql = require("./graphql");
+const HttpSessions = require("./httpSessions");
+const Network = require("./network");
+const Openapi = require("./openapi");
+const Params = require("./params");
+const Pnh = require("./pnh");
+const Pscan = require("./pscan");
+const Reports = require("./reports");
+const Replacer = require("./replacer");
+const Reveal = require("./reveal");
+const Retest = require("./retest");
+const Revisit = require("./revisit");
+const RuleConfig = require("./ruleConfig");
+const Script = require("./script");
+const Search = require("./search");
+const Selenium = require("./selenium");
+const SessionManagement = require("./sessionManagement");
+const Soap = require("./soap");
+const Spider = require("./spider");
+const Stats = require("./stats");
+const Users = require("./users");
+const Wappalyzer = require("./wappalyzer");
+const Websocket = require("./websocket");
+
 
 // base JSON api url
-const BASE = 'http://zap/JSON';
+//const BASE = "http://zap/JSON";
 // base OTHER api url
-const BASE_OTHER = 'http://zap/OTHER';
+//const BASE_OTHER = "http://zap/OTHER";
 
 function ClientApi(options) {
-  const requestOptions = {
-    proxy: { ...{ proxy: 'http://127.0.0.1:8080' }, ...options }.proxy,
-    method: 'GET',
-    json: true,
-    headers: options.apiKey ? { 'X-ZAP-API-Key': options.apiKey } : {}
+  // const requestOptions = {
+  //   proxy: { ...{ proxy: "http://127.0.0.1:8080" }, ...options }.proxy,
+  //   method: "GET",
+  //   json: true,
+  //   headers: options.apiKey ? { "X-ZAP-API-Key": options.apiKey } : {},
+  // };
+
+  requestConfig = {
```

Comment:
This should set the `proxy` configuration and use the `zap` domain for the `baseURL`.